python/lsst/ts/hvac/hvac_csc.py

```diff
@@ -0,0 +1,283 @@
+        """
+        self.log.info("Disconnecting")
+        if self.telemetry_task:
+            self.telemetry_task.cancel()
+        if self.connected:
+            self.mqtt_client.disconnect()
+
+    async def handle_summary_state(self):
+        """Override of the handle_summary_state function to connect or
+        disconnect to the HVAC server (or the mock server) when needed.
+        """
+        self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
+        if self.disabled_or_enabled:
+            if not self.connected:
+                await self.connect()
+        else:
+            await self.disconnect()
+
+    async def configure(self, config):
+        self.config = config
+
+    async def implement_simulation_mode(self, simulation_mode):
+        if simulation_mode not in (0, 1):
+            raise salobj.ExpectedError(f"Simulation_mode={simulation_mode} must be 0 or 1")
+
+    async def do_chiller01P01(self, data):
+        self.assert_enabled()
+        await self.mqtt_client.chiller01P01(
+            setpoint_activo=data.setpointActivo, comando_encendido=data.comandoEncendido
+        )
+
+    async def do_crack01P02(self, data):
+        self.assert_enabled()
+        await self.mqtt_client.crack01P02(
+            setpoint_humidificador=data.setpointHumidificador,
+            setpoint_deshumidificador=data.setpointDeshumidificador,
+            set_point_cooling=data.setPointCooling,
+            set_point_heating=data.setPointHeating,
+            comando_encendido=data.comandoEncendido,
+        )
+
+    async def do_fancoil01P02(self, data):
+        self.assert_enabled()
+        await self.mqtt_client.fancoil01P02(
+            perc_apertura_valvula_frio=data.percAperturaValvulaFrio,
+            setpoint_cooling_day=data.setpointCoolingDay,
+            setpoint_heating_day=data.setpointHeatingDay,
+            setpoint_cooling_night=data.setpointCoolingNight,
+            setpoint_heating_night=data.setpointHeatingNight,
+            setpoint_trabajo=data.setpointTrabajo,
+            comando_encendido=data.comandoEncendido,
+        )
+
+    async def do_manejadoraLower01P05(self, data):
+        self.assert_enabled()
+        await self.mqtt_client.manejadoraLower01P05(
+            setpoint_trabajo=data.setpointTrabajo,
+            setpoint_ventilador_min=data.setpointVentiladorMin,
+            setpoint_ventilador_max=data.setpointVentiladorMax,
+            temperatura_anticongelante=data.temperaturaAnticongelante,
+            comando_encendido=data.comandoEncendido,
+        )
+
+    async def do_manejadoraSblancaP04(self, data):
+        self.assert_enabled()
+        await self.mqtt_client.manejadoraSblancaP04(
+            valor_consigna=data.valorConsigna,
+            setpoint_ventilador_min=data.setpointVentiladorMin,
+            setpoint_ventilador_max=data.setpointVentiladorMax,
+            comando_encendido=data.comandoEncendido,
+        )
```

Comment:
This basically is why I explicitly wanted you to do the review. I tried various ways of eliminating boilerplate but couldn't find a proper one that worked. I simply still am too inexperienced in Python. Perhaps I should have asked you before the review to avoid this but that proved hard with my other tasks and the big time difference.

Reply:
I will think this over and *may* change this at a later stage.